h2/src/test/org/h2/test/testScript.sql

```diff
@@ -3196,6 +3196,18 @@ drop domain email;
 drop domain gmail;
 > ok
 
+CREATE VALUE TYPE person FOR "org.h2.samples.CustomTypes$PersonType" WITH param1,param2;
+> ok
+
+select CLASS_NAME, SQL from information_schema.custom_types;
+> CLASS_NAME                            SQL
+> ------------------------------------- ----------------------------------------------------------------------------------------
+> org.h2.samples.CustomTypes$PersonType CREATE VALUE TYPE PERSON FOR "org.h2.samples.CustomTypes$PersonType" WITH PARAM1, PARAM2
+> rows: 1
+
+drop VALUE TYPE person;
+> ok
+
 create force view address_view as select * from address;
 > ok
 
```

Comment:
The thing is, this cast (the part afrer `AS`, to be precise) is parsed exactly like any column in `CREATE TABLE` and actually even a `Column` object is created for a cast under the hood, so to me it is looks legit. Column knows what custom type is associated with it and can perform needed conversions transparently, without need for anyone from outside to bother about any new types. Do you have more concrete arguments why this binding is bad?